src/Compilers/Core/Desktop/AbstractAnalyzerAssemblyLoader.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    public abstract class AbstractAnalyzerAssemblyLoader : IAnalyzerAssemblyLoader
```

Comment:
@jaredpar I could make all of these types internal by pushing them out of `CodeAnalysis.Desktop` and into the individual hosts (csc, vbc, vbcscompiler, VS, etc.). The reason I didn't do that is that `AbstractAnalyzerAssemblyLoader` uses the metadata reader to check the assembly's identity before loading it. So I have two options:

1. Take a dependency on the System.Reflection.Metadata package in each host that doesn't currently (csc, vbc, csi, vbi).
2. Drop the identity checking, and load assemblies without double-checking the identity.

I don't have a strong preference between the two of these.